win32/agent.py

```diff
@@ -145,14 +151,21 @@ def is_alive(self):
     def is_enabled(self):
         return self._process.is_enabled
 
+    def _can_restart(self):
+        now = time.time()
+        while(self._restarts and self._restarts[0] < now - self._RESTART_TIMEFRAME):
+            self._restarts.popleft()
+
+        return len(self._restarts) < self._MAX_RESTARTS
+
     def restart(self):
-        self._count_restarts += 1
-        if self._count_restarts >= self._MAX_RESTARTS:
+        if not self._can_restart():
             servicemanager.LogInfoMsg(
                 "%s reached the limit of restarts. Not restarting..." % self._name)
```

Comment:
To ease a potential debug, could you add the number of restarts (or tries), max number of restarts, and maybe the time window ? 
So something like:
```
"{0} reached the limit of restarts ({1} tries during the last {3}s (max authorized: {2})). Not restarting...".format(self._name, len(self._restarts), self._restarts[0] - time.time()) ?
```